script/gen_requirements_all.py

```diff
@@ -228,21 +229,31 @@ def gather_modules():
                 errors.append(package)
             continue
 
-        if not getattr(module, 'REQUIREMENTS', None):
+        manifest_path = os.path.join(
+            os.path.dirname(module.__file__),
+            'manifest.json'
+        )
+
+        manifest_file_exists = os.path.isfile(manifest_path)
+        if (not manifest_file_exists
```

Comment:
raise error if manifest file does not exist

Reply:
We are not just iterating through components, we also iterate through scripts/auth. 

I think that we should split this function in 2. One doing the old way for scripts/auth and one for iterating on manifests. We can use `iter_manifests` from `script/manifests/manifest_helper.py`. 

Can we also move this method to `script/manifest`, to indicate that it relies on manifests ?